applications/FluidDynamicsHydraulicsApplication/custom_utilities/hydraulic_fluid_auxiliary_utilities.cpp

```diff
@@ -302,9 +302,141 @@ void HydraulicFluidAuxiliaryUtilities::SetInletFreeSurface(ModelPart &rModelPart
         if (rNode.Is(rSkinFlag)){
             double inlet_dist = rNode.GetValue(rDistanceVariable);
             rNode.FastGetSolutionStepValue(DISTANCE) = inlet_dist;
-            rNode.Fix(DISTANCE);
+            // The distance is fixed in the water nodes since the inlet has velocity.
+            if (inlet_dist<0.0){
+                rNode.Fix(DISTANCE);
+            }
+        }
+    });
+}
+
+void HydraulicFluidAuxiliaryUtilities::TurnOffGravityOnAirElements(ModelPart &rModelPart){
+    block_for_each(rModelPart.Elements(), [](Element &rElement){
+        auto &r_geom = rElement.GetGeometry();
+        Vector distances = ZeroVector(r_geom.PointsNumber());
+        Vector gravity = ZeroVector(3);
+        for (unsigned int i_nodes = 0; i_nodes < r_geom.PointsNumber(); i_nodes++){
+            distances[i_nodes] = r_geom[i_nodes].FastGetSolutionStepValue(DISTANCE);
+            }
+        if (FluidAuxiliaryUtilities::IsPositive(distances)){
+            for (unsigned int i_nodes = 0; i_nodes < r_geom.PointsNumber(); i_nodes++)
+            {
+                r_geom[i_nodes].FastGetSolutionStepValue(BODY_FORCE) = gravity;
+            }
+        }
+    });
+}
+
+void HydraulicFluidAuxiliaryUtilities::FixCornerNodeVelocity(ModelPart &rModelPart, double MaximumAngle)
```

Comment:
```suggestion
void HydraulicFluidAuxiliaryUtilities::FixCornerNodeVelocity(
    ModelPart &rModelPart,
    const double MaximumAngle)
```
Please follow the [style guide](https://github.com/KratosMultiphysics/Kratos/wiki/Style-Guide).